# Analysis and Classification Model of the Wine Dataset

In [3]:
# Imports
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [4]:
address = '/Users/phillipgregory/Documents/Career/Data science/beginner_datasets/wine.csv'
df = pd.read_csv(address)
df.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality type  
0      9.4        5  red  
1      9.8        5  red  
2      9.8        5  red  
3      9.8        6  red  
4      9.4        5  red

## Check Data Quality

In [5]:
# Null values
df.isnull().sum()

fixed acidity           0
volatile acidity        0
citric acid             0
residual sugar          0
chlorides               0
free sulfur dioxide     0
total sulfur dioxide    0
density                 0
pH                      0
sulphates               0
alcohol                 0
quality                 0
type                    0
dtype: int64

In [6]:
# Check data types
df.dtypes

fixed acidity           float64
volatile acidity        float64
citric acid             float64
residual sugar          float64
chlorides               float64
free sulfur dioxide     float64
total sulfur dioxide    float64
density                 float64
pH                      float64
sulphates               float64
alcohol                 float64
quality                   int64
type                     object
dtype: object

In [7]:
# Check df shape
df.shape

(6497, 13)

## Logistic Regression
* I will use logistic regression to classify the type of wine: Red or White

In [17]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier

In [10]:
# Columns for prediction
X = df.drop('type', axis=1)
y = df.type

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

In [16]:
# Instantiate model
clf = LogisticRegression(max_iter=5000)

# Fit model on the training data
clf.fit(X_train, y_train)

LogisticRegression(max_iter=5000)

In [18]:
# Make predictions
y_pred = clf.predict(X_test)

In [21]:
# Test Logistic regression model performance
from sklearn.metrics import accuracy_score

print("Accuracy score: %.3f" % accuracy_score(y_pred, y_test))

Accuracy score: 0.984


In [23]:
# Test Random Forest Classifier
clf1 = RandomForestClassifier()

clf1.fit(X_train, y_train)

y_pred1 = clf1.predict(X_test)

print("Accuracy score: %.3f" % accuracy_score(y_pred1, y_test))

Accuracy score: 0.995


## Conclusion
The Random Forest Classifier has a higher accuracy than Logistic Regression at 0.995 vs 0.984.